src/Metadata/Resource/ResourceNameCollection.php

```diff
@@ -19,13 +19,15 @@
 final class ResourceNameCollection implements \IteratorAggregate, \Countable
 {
     private $classes;
+    private $context;
 
     /**
      * @param string[] $classes
      */
-    public function __construct(array $classes = [])
+    public function __construct(array $classes = [], array $context = [])
```

Comment:
This hackish solution needs to go. (Yeah, I know it's a WIP. Just pointing it out...)